--- a/net/minecraft/client/Minecraft.java
+++ b/net/minecraft/client/Minecraft.java
@@ -881,6 +881,10 @@
             }
         }
     }
+    
+    public CompletableFuture<Void> delayTextureReload() {
+        return this.<CompletableFuture<Void>>submit(() -> reloadResourcePacks()).thenCompose(completableFuture -> completableFuture);
+    }
 
     private void selfTest() {
         boolean flag = false;
@@ -2326,10 +2330,6 @@
         return instance;
     }
 
-    public CompletableFuture<Void> delayTextureReload() {
-        return this.<CompletableFuture<Void>>submit(this::reloadResourcePacks).thenCompose(completableFuture -> completableFuture);
-    }
-
     public void updateReportEnvironment(ReportEnvironment reportEnvironment) {
         if (!this.reportingContext.matches(reportEnvironment)) {
             this.reportingContext = ReportingContext.create(reportEnvironment, this.userApiService);
