--- a/net/minecraft/client/OptionInstance.java
+++ b/net/minecraft/client/OptionInstance.java
@@ -287,7 +287,7 @@
             return new OptionInstance.SliderableValueSet<R>() {
                 @Override
                 public Optional<R> validateValue(R value) {
-                    return IntRangeBase.this.validateValue((T)Integer.valueOf(toIntFunction.applyAsInt(value))).map(intFunction::apply);
+                    return IntRangeBase.this.validateValue(Integer.valueOf(toIntFunction.applyAsInt(value))).map(intFunction::apply);
                 }
 
                 @Override
