--- a/net/minecraft/client/multiplayer/ClientPacketListener.java
+++ b/net/minecraft/client/multiplayer/ClientPacketListener.java
@@ -1245,8 +1245,8 @@
     public void handleOpenSignEditor(ClientboundOpenSignEditorPacket packet) {
         PacketUtils.ensureRunningOnSameThread(packet, this, this.minecraft);
         BlockPos pos = packet.getPos();
-        BlockEntity blockState = this.level.getBlockEntity(pos);
-        if (blockState instanceof SignBlockEntity signBlockEntity) {
+        BlockEntity blockEntity = this.level.getBlockEntity(pos);
+        if (blockEntity instanceof SignBlockEntity signBlockEntity) {
             this.minecraft.player.openTextEdit(signBlockEntity, packet.isFrontText());
         } else {
             BlockState blockState = this.level.getBlockState(pos);
@@ -1565,7 +1565,7 @@
                 .<T>registry(registry)
                 .orElseThrow(() -> new IllegalStateException("Unknown registry " + registry));
             Map<TagKey<T>, List<Holder<T>>> map = new HashMap<>();
-            TagNetworkSerialization.deserializeTagsFromNetwork(registry, registry1, networkPayload, map::put);
+            TagNetworkSerialization.deserializeTagsFromNetwork((ResourceKey<? extends Registry<T>>) registry, registry1, networkPayload, map::put);
             registry1.bindTags(map);
         }
     }
