--- a/net/minecraft/client/telemetry/TelemetryPropertyMap.java
+++ b/net/minecraft/client/telemetry/TelemetryPropertyMap.java
@@ -59,7 +59,7 @@
                     T object = mapLike.get(telemetryProperty.id());
                     if (object != null) {
                         DataResult<V> dataResult1 = telemetryProperty.codec().parse(dynamicOps, object);
-                        return dataResult.apply2stable((builder, object1) -> builder.put(telemetryProperty, (T)object1), dataResult1);
+                        return dataResult.apply2stable((builder, object1) -> builder.put((TelemetryProperty<T>)telemetryProperty, (T)object1), dataResult1);
                     } else {
                         return dataResult;
                     }
