--- a/net/minecraft/client/gui/screens/achievement/StatsScreen.java
+++ b/net/minecraft/client/gui/screens/achievement/StatsScreen.java
@@ -458,11 +458,11 @@
                     i = 0;
                     i1 = 0;
                 } else if (ItemStatisticsList.this.blockColumns.contains(ItemStatisticsList.this.sortColumn)) {
-                    StatType<Block> statType = ItemStatisticsList.this.sortColumn;
+                    StatType<Block> statType = (StatType<Block>) ItemStatisticsList.this.sortColumn;
                     i = item instanceof BlockItem ? StatsScreen.this.stats.getValue(statType, ((BlockItem)item).getBlock()) : -1;
                     i1 = item1 instanceof BlockItem ? StatsScreen.this.stats.getValue(statType, ((BlockItem)item1).getBlock()) : -1;
                 } else {
-                    StatType<Item> statType = ItemStatisticsList.this.sortColumn;
+                    StatType<Item> statType = (StatType<Item>) ItemStatisticsList.this.sortColumn;
                     i = StatsScreen.this.stats.getValue(statType, item);
                     i1 = StatsScreen.this.stats.getValue(statType, item1);
                 }
