--- a/net/minecraft/client/gui/Gui.java
+++ b/net/minecraft/client/gui/Gui.java
@@ -410,8 +410,8 @@
     protected void renderEffects(GuiGraphics guiGraphics) {
         Collection<MobEffectInstance> activeEffects = this.minecraft.player.getActiveEffects();
         if (!activeEffects.isEmpty()) {
-            Screen i1 = this.minecraft.screen;
-            if (i1 instanceof EffectRenderingInventoryScreen effectRenderingInventoryScreen && effectRenderingInventoryScreen.canSeeEffects()) {
+            Screen screen = this.minecraft.screen;
+            if (screen instanceof EffectRenderingInventoryScreen effectRenderingInventoryScreen && effectRenderingInventoryScreen.canSeeEffects()) {
                 return;
             }
 
@@ -455,9 +455,10 @@
                     TextureAtlasSprite textureAtlasSprite = mobEffectTextures.get(effect);
                     int i5 = i3;
                     float f1 = f;
+                    int finalI = i2;
                     list.add(() -> {
                         guiGraphics.setColor(1.0F, 1.0F, 1.0F, f1);
-                        guiGraphics.blit(i2 + 3, i5 + 3, 0, 18, 18, textureAtlasSprite);
+                        guiGraphics.blit(finalI + 3, i5 + 3, 0, 18, 18, textureAtlasSprite);
                         guiGraphics.setColor(1.0F, 1.0F, 1.0F, 1.0F);
                     });
                 }
