--- a/com/mojang/realmsclient/gui/screens/RealmsUploadScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsUploadScreen.java
@@ -237,8 +237,8 @@
                             } else {
                                 uploadInfo = null;
         
-                                int i;
-                                for(i = 0; i < 20; ++i) {
+                                
+                                for(int i = 0; i < 20; ++i) {
                                     try {
                                         if (this.cancelled) {
                                             this.uploadCancelled();
@@ -263,7 +263,7 @@
                                     } else if (this.cancelled) {
                                         this.uploadCancelled();
                                     } else {
-                                        i = (int)(new File(this.minecraft.gameDirectory.getAbsolutePath(), "saves"));
+                                        File i = (new File(this.minecraft.gameDirectory.getAbsolutePath(), "saves"));
                                         file = this.tarGzipArchive(new File(i, this.selectedLevel.getLevelId()));
                                         if (this.cancelled) {
                                             this.uploadCancelled();
@@ -331,13 +331,10 @@
                                 }
                             }
                         } catch (IOException var21) {
-                            uploadInfo = var21;
                             this.setErrorMessage(Component.translatable("mco.upload.failed", var21.getMessage()));
                         } catch (RealmsServiceException var22) {
-                            uploadInfo = var22;
                             this.setErrorMessage(Component.translatable("mco.upload.failed", var22.toString()));
                         } catch (InterruptedException var23) {
-                            uploadInfo = var23;
                             LOGGER.error("Could not acquire upload lock");
                         }
                     } finally {
