--- a/com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen.java
@@ -207,36 +207,43 @@
                     WorldTemplatePaginatedList worldTemplatePaginatedList = output;
     
                     Either<WorldTemplatePaginatedList, String> either;
-                    for(RealmsClient realmsClient = RealmsClient.create();
-                        worldTemplatePaginatedList != null;
+                    RealmsClient realmsClient = RealmsClient.create();
+                    while (worldTemplatePaginatedList != null) {
+                       either = RealmsSelectWorldTemplateScreen.this.fetchTemplates(worldTemplatePaginatedList, realmsClient);
+                        Either<WorldTemplatePaginatedList, String> finalEither = either;
                         worldTemplatePaginatedList = RealmsSelectWorldTemplateScreen.this.minecraft
                             .submit(
                                 () -> {
-                                    if (either.right().isPresent()) {
-                                        RealmsSelectWorldTemplateScreen.LOGGER.error("Couldn't fetch templates: {}", either.right().get());
+                                    if (finalEither.right().isPresent()) {
+                                        RealmsSelectWorldTemplateScreen.LOGGER.error(
+                                            "Couldn't fetch templates: {}", finalEither.right().get());
                                         if (RealmsSelectWorldTemplateScreen.this.worldTemplateObjectSelectionList.isEmpty()) {
                                             RealmsSelectWorldTemplateScreen.this.noTemplatesMessage = TextRenderingUtils.decompose(
                                                 I18n.get("mco.template.select.failure")
                                             );
                                         }
-                
+
                                         return null;
                                     } else {
-                                        WorldTemplatePaginatedList worldTemplatePaginatedList1 = either.left().get();
-                
-                                        for(WorldTemplate worldTemplate : worldTemplatePaginatedList1.templates) {
-                                            RealmsSelectWorldTemplateScreen.this.worldTemplateObjectSelectionList.addEntry(worldTemplate);
+                                        WorldTemplatePaginatedList worldTemplatePaginatedList1 = finalEither.left()
+                                            .get();
+
+                                        for (WorldTemplate worldTemplate : worldTemplatePaginatedList1.templates) {
+                                            RealmsSelectWorldTemplateScreen.this.worldTemplateObjectSelectionList.addEntry(
+                                                worldTemplate);
                                         }
-                
+
                                         if (worldTemplatePaginatedList1.templates.isEmpty()) {
                                             if (RealmsSelectWorldTemplateScreen.this.worldTemplateObjectSelectionList.isEmpty()) {
-                                                String string = I18n.get("mco.template.select.none", "%link");
+                                                String string = I18n.get("mco.template.select.none",
+                                                    "%link");
                                                 TextRenderingUtils.LineSegment lineSegment = TextRenderingUtils.LineSegment.link(
-                                                    I18n.get("mco.template.select.none.linkTitle"), "https://aka.ms/MinecraftRealmsContentCreator"
+                                                    I18n.get("mco.template.select.none.linkTitle"),
+                                                    "https://aka.ms/MinecraftRealmsContentCreator"
                                                 );
-                                                RealmsSelectWorldTemplateScreen.this.noTemplatesMessage = TextRenderingUtils.decompose(string, lineSegment);
+                                                RealmsSelectWorldTemplateScreen.this.noTemplatesMessage = TextRenderingUtils.decompose(
+                                                    string, lineSegment);
                                             }
-                
                                             return null;
                                         } else {
                                             return worldTemplatePaginatedList1;
@@ -244,9 +251,7 @@
                                     }
                                 }
                             )
-                            .join()
-                    ) {
-                        either = RealmsSelectWorldTemplateScreen.this.fetchTemplates(worldTemplatePaginatedList, realmsClient);
+                            .join();
                     }
                 }
             })
