--- a/net/minecraft/client/gui/components/CycleButton.java
+++ b/net/minecraft/client/gui/components/CycleButton.java
@@ -151,11 +151,11 @@
     }
 
     public static CycleButton.Builder<Boolean> booleanBuilder(Component componentOn, Component componentOff) {
-        return new CycleButton.Builder<>(value -> value ? componentOn : componentOff).withValues(BOOLEAN_OPTIONS);
+        return new CycleButton.Builder<Boolean>(value -> value ? componentOn : componentOff).withValues(BOOLEAN_OPTIONS);
     }
 
     public static CycleButton.Builder<Boolean> onOffBuilder() {
-        return new CycleButton.Builder<>(value -> value ? CommonComponents.OPTION_ON : CommonComponents.OPTION_OFF).withValues(BOOLEAN_OPTIONS);
+        return new CycleButton.Builder<Boolean>(value -> value ? CommonComponents.OPTION_ON : CommonComponents.OPTION_OFF).withValues(BOOLEAN_OPTIONS);
     }
 
     public static CycleButton.Builder<Boolean> onOffBuilder(boolean initialValue) {
